metadata-ingestion/src/datahub/ingestion/source/fivetran/fivetran_log_api.py

```diff
@@ -0,0 +1,123 @@
+import json
+import logging
+from typing import Any, Dict, List, Tuple
+
+from sqlalchemy import create_engine
+
+from datahub.ingestion.source.fivetran.config import Constant, FivetranLogConfig
+from datahub.ingestion.source.fivetran.data_classes import Connector, Job
+from datahub.ingestion.source.fivetran.fivetran_query import FivetranLogQuery
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class FivetranLogAPI:
+    def __init__(self, fivetran_log_config: FivetranLogConfig) -> None:
+        self.fivetran_log_config = fivetran_log_config
+        self.engine = self._get_log_destination_engine()
+
+    def _get_log_destination_engine(self) -> Any:
+        destination_platform = self.fivetran_log_config.destination_platform
+        engine = None
+        if destination_platform == "snowflake":
```

Comment:
as long as the config type has a get_sql_alchemy_url and get_options method, we should be able to call create_engine() right? what's the reason we need to do this per destination platform?

Reply:
Yes, As creation of sqlalchemy engine using sqlalchemy URL to execute the queries is the common thing for other destination as well. Hence, surely we will need one common method that is get_sql_alchemy_url.
We can remove get_options method as its not required compulsary.